src/PrestaShopBundle/Resources/config/services/core/grid.yml

```diff
@@ -169,6 +183,14 @@ services:
             - "@=service('router').generate('admin_contacts_index')"
         public: true
 
+    prestashop.core.grid.definition.factory.language:
+        class: 'PrestaShop\PrestaShop\Core\Grid\Definition\Factory\LanguageGridDefinitionFactory'
+        parent: 'prestashop.core.grid.definition.factory.abstract_grid_definition'
+        arguments:
+            - "@=service('router').generate('admin_common_reset_search', {'controller': 'language', 'action': 'index'})"
```

Comment:
😅 I dont know whether this is right
We have a service definition that relies on:
- the router method (ok, this one is not going to change a lot)
- a route configuration, from a YAML file

I'm afraid few people, if they modify a routing YAML file, will think "ok, I got to modify the service definition file too".
It consequently looks fragile to me.

However if we can make sure a test will find out if we break one service definition, I'm ok 😀 

Here's one idea (however feel free to suggest others): we could create a test suite that basically parses the Symfony Container (probably through a Symfony Command) and tries to get **every** service. If the service definition is broken, this test would fail.

It would look like this:
```
$errors = [];
$container = $this->getContainer();
foreach($container->getAllServices as $serviceName) {
    try {
        $myService = $container->get($serviceName);
    } catch (ServiceCannotBeBuiltException $e) {
        $errors[] = $e;
    }
}

if (!empty($errors)) {
    // test has fail, print the list of services which cannot be built with the reasons
}
```

However this would only work for public services. Need to find an idea for private services too 😉

Reply:
actually i noticed same issue and also the problem that you have to inject these params every time you create grid definition. So basically, what i did is updated `SearchAndResetType` as it's the reason why we need to inject these params in the first place.

You can check at this line https://github.com/PrestaShop/PrestaShop/pull/10916/files#diff-74412ed3073d7a554856e4bdf755a4efR315 wdyt?

Reply:
I'm really not a fan of passing `reset_route_params`. I'm afraid we start adding more and more array-driven configuration options until we get something like [this legacy controller config](https://github.com/PrestaShop/PrestaShop/blob/develop/controllers/admin/AdminEmployeesController.php#L106). Big config arrays that look terrifying for new prestashop developers as they think "I have to pass ALL these options ? what are them ? what are possible values ?" and if they input the wrong values, the BO fails.

I might be traumatized by previous experiences 😅 so let's ask around for opinions
@Quetzacoalt91  @PierreRambaud @jolelievre

Reply:
thats a valid point.